pallets/bucket-nfts/src/mock.rs

```diff
@@ -367,44 +348,34 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
     ext
 }
 
-// Converter from the ThresholdType type (FixedU128) to the BlockNumber type (u64).
+// Converter from the ThresholdType to the BlockNumber type and vice versa.
 // It performs a saturated conversion, so that the result is always a valid BlockNumber.
-pub struct SaturatingThresholdTypeToBlockNumberConverter;
+pub struct ThresholdTypeToBlockNumberConverter;
 
-impl Convert<ThresholdType, BlockNumberFor<Test>>
-    for SaturatingThresholdTypeToBlockNumberConverter
-{
+impl Convert<ThresholdType, BlockNumberFor<Test>> for ThresholdTypeToBlockNumberConverter {
     fn convert(threshold: ThresholdType) -> BlockNumberFor<Test> {
-        (threshold.into_inner() / FixedU128::accuracy()).saturated_into()
+        threshold.saturated_into()
     }
 }
 
-// Converter from the BlockNumber type (u64) to the ThresholdType type (FixedU128).
-pub struct BlockNumberToThresholdTypeConverter;
-
-impl Convert<BlockNumberFor<Test>, ThresholdType> for BlockNumberToThresholdTypeConverter {
-    fn convert(block_number: BlockNumberFor<Test>) -> ThresholdType {
-        FixedU128::from_inner((block_number as u128) * FixedU128::accuracy())
+impl ConvertBack<ThresholdType, BlockNumberFor<Test>> for ThresholdTypeToBlockNumberConverter {
+    fn convert_back(block_number: BlockNumberFor<Test>) -> ThresholdType {
+        block_number.saturated_into()
     }
 }
 
-// Converter from the Hash type from the runtime (BlakeTwo256) to the ThresholdType type (FixedU128).
-// Since we can't convert directly a hash to a FixedU128 (since the hash type used in the runtime has
-// 256 bits and FixedU128 has 128 bits), we convert the hash to a BigUint, then map it to a FixedU128
-// by keeping its relative position between zero and the maximum 256-bit number.
+/// Converter from the [`Hash`] type to the [`ThresholdType`].
 pub struct HashToThresholdTypeConverter;
 impl Convert<<Test as frame_system::Config>::Hash, ThresholdType> for HashToThresholdTypeConverter {
     fn convert(hash: <Test as frame_system::Config>::Hash) -> ThresholdType {
         // Get the hash as bytes
         let hash_bytes = hash.as_ref();
 
-        // Get the 16 least significant bytes of the hash and interpret them as a u128
-        let truncated_hash_bytes: [u8; 16] =
-            hash_bytes[16..].try_into().expect("Hash is 32 bytes; qed");
-        let hash_as_u128 = u128::from_be_bytes(truncated_hash_bytes);
+        // Get the 4 least significant bytes of the hash and interpret them as an u32
+        let truncated_hash_bytes: [u8; 4] =
+            hash_bytes[28..].try_into().expect("Hash is 32 bytes; qed");
```

Comment:
Check the 28 here too.

Reply:
Resolving discussion in duplicate comment.